### PR TITLE
Tests are now involved in all this

### PR DESCRIPTION
**Pull request by pastjean**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,7 +20,10 @@
     "ava": "^0.14.0",
     "es6-promise": "3.1.2",
     "exports-loader": "0.6.3",
+    "express": "^4.13.4",
     "isomorphic-fetch": "2.2.1",
+    "jsdom": "^8.4.0",
+    "sinon": "^1.17.3",
```
Comment (GermainBergeron) [inline package.json]: Remove `^` in version numbers

File: test/analytics_test.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import test from 'ava';
+import * as analytics from '../src/analytics';
+
+import * as express from 'express';
+import * as http from 'http';
+
+var app:express.Application = express();
```
Comment (GermainBergeron) [inline test/analytics_test.ts]: missing space after `:`

Comment (GermainBergeron) [inline test/analytics_test.ts]: You should consider linting the tests as well ;)

File: test/analytics_test.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const server: http.Server = (<any>http).createServer(app).listen()
+app.set('port', server.address().port);
+
+test('Analytics: can post a view event', t => {
+
+    const response : analytics.ViewEventResponse = {
+        raw:undefined,
+        visitId : '123',
+        visitorId: '213'
```
Comment (GermainBergeron) [inline test/analytics_test.ts]: choose a formating style !

File: test/analytics_test.ts (hunk 2 above)
Comment (wfortin) [inline test/analytics_test.ts]: Formatting here. You should setup tslint 😄 !

File: test/simpleanalytics_test.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+test('SimpleAnalytics: can send pageview', t => {
+    simpleanalytics('init','MYTOKEN', `http://localhost:${server.address().port}`)
+    simpleanalytics('send','pageview')
+});
+
+
+test('SimpleAnalytics: can send pageview with customdata', t => {    
+    simpleanalytics('init','MYTOKEN', `http://localhost:${server.address().port}`)
+    simpleanalytics('send','pageview', {somedata:'asd'});
+});
```
Comment (wfortin) [inline test/simpleanalytics_test.ts]: The file formatting is weird. Have a single blank space between each test. Sometimes you have 0 sometimes 1 and then 2...

**Comment by wfortin** [comment]
Great job on setting up AVA !

**Comment by GermainBergeron** [comment]
LGTM! Well done for the tests with express

Maybe you could setup code coverage? Everybody love 'em git badge
You should also run the tests when building with Travis

File: test/analytics_test.ts (hunk 2 above)
Comment (pastjean) [inline test/analytics_test.ts]: j'ai ajouté les tests dans le linting :D ça va régler tout ça

**Comment by pastjean** [comment]
@GermainBergeron je vois ici un germain qui veut un template pour des tests et du coverage avec ava :P , je vais voir si je suis capable de faire ça vite

File: src/analytics.ts [hunk 5]
```diff
@@ -23,7 +23,14 @@ export class Client {
     private endpoint: string;
     private token: string;
 
-    constructor(opts: ClientOptions = {token: ''} ) {
+    constructor(opts: ClientOptions) {
+        if (typeof opts === 'undefined') {
```
Comment (JPLachance) [inline src/analytics.ts]: Underscore.js is useful for those comparison. Did you think about using it?

File: tsconfig.test.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+        "noImplicitAny": true,
+        "removeComments": true,
+        "sourceMap": true
+    },
+    "files": [
+      "test/lib.d.ts",
+      "test/simpleanalytics_test.ts",
+      "test/analytics_test.ts",
+      "typings/main.d.ts"
+      ]
```
Comment (JPLachance) [inline tsconfig.test.json]: Formatting in that file

File: src/analytics.ts (hunk 5 above)
Comment (pastjean) [inline src/analytics.ts]: one less lib to include in. i'm trying to keep the builded artifact lightweight

Comment (wfortin) [inline src/analytics.ts]: +1 for keeping things small.

**Pull request merged**